packages/core/src/modules/connections/protocols/trust-ping/v2/handlers/TrustPingMessageHandler.ts

```diff
@@ -16,11 +18,27 @@ export class TrustPingMessageHandler implements MessageHandler {
   }
 
   public async handle(messageContext: MessageHandlerInboundMessage<TrustPingMessageHandler>) {
+    let connection: ConnectionRecord | undefined = messageContext.connection
```

Comment:
If we want to create a connection on receiving any type of DidComm V2 message we can move this logic to `MessageReceiver processUnpackedMessage / findConnection` method. 
Making a connection on Ping is not a standard but seems to be more natural for me..

Reply:
I agree on creating the connection upon the reception of the first message addressed to the DID. Especially if we want to have the concept of 'Connection' when connecting to agents not based on AFJ, as they might not send any trust ping at all.

Reply:
Moved